### ISSUE TITLE
[java] MissingOverride false-positive triggered by package-private method overwritten in another package by extending class. 

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#1969]
**Affects PMD Version:** 
6.16.0, 6.17.0 

**Rule:**
[MissingOverride](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#missingoverride)

**Description:**
False positive for package-private method triggered by class extended in another package. 
A package-private method cannot
be directly overridden by a method in a different package [JLS 8.4.8.1](https://docs.oracle.com/javase/specs/jls/se12/html/jls-8.html#jls-8.4.8.1). 
The method in the derived class is not the inherited method and thus the rule should not flag it as needing an @Override annotation. Package-private methods can only be overridden by classes defined in the same package as the base class.

**Code Sample demonstrating the issue:**

``` java
package click;
public class CodeTalk {
    public void doIt() {
        printMessage();
    }
    void printMessage() {
        System.out.println("Click");
    }
}

package hack;
import click.CodeTalk;
public class TypeIt {
    private static class ClickIt extends CodeTalk {
        void printMessage() {
            System.out.println("Hack");
        }
    }

    public static void main(String[] args) {
        ClickIt clickit = new ClickIt();
        clickit.doIt();
    }
}
```

**Running PMD through:** *[CLI | Ant] *

**Comment by adangel** [comment]
Fixed via #2899 for PMD 7.
